File: queryable-encryption/go/aws/reader/your-credentials.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
)

var credentials = map[string]string{
	// Mongo Paths + URI
	"MONGODB_URI":     os.Getenv("MONGODB_URI"),
	"SHARED_LIB_PATH": os.Getenv("SHARED_LIB_PATH"),
	// AWS Credentials
	"AWS_ACCESS_KEY_ID":     os.Getenv("AWS_ACCESS_KEY_ID"),
	"AWS_SECRET_ACCESS_KEY": os.Getenv("AWS_SECRET_ACCESS_KEY"),
	"AWS_KEY_REGION":        os.Getenv("AWS_KEY_REGION"),
	"AWS_KEY_ARN":           os.Getenv("AWS_KEY_ARN"),
}

// check if credentials object contains placeholder values
func check_for_placeholders() {
	var error_buffer []string
	for key, value := range credentials {
		// check for placeholder text
		if value == "" {
			error_message := fmt.Sprintf("You must fill out the %s environment variable.", key)
			error_buffer = append(error_buffer, error_message)
		}
	}
	// raise an error if errors in buffer
	if len(error_buffer) > 0 {
		message := strings.Join(error_buffer[:], "\n")
		log.Fatal(message)
	}
}

// return credentials object and ensure it has been populated
func GetCredentials() map[string]string {
	check_for_placeholders()
	return credentials
}

File: queryable-encryption/mongosh/gcp/reader/your_credentials.js
/*
return credentials object and ensure it has been populated
**/
function getCredentials() {
  checkForPlaceholders();
  return credentials;
}

const credentials = {
  // Mongo Paths + URI
  MONGODB_URI: "<your MongoDB URI here>",

  // GCP Credentials
  GCP_EMAIL: "<your GCP email>",
  GCP_PRIVATE_KEY: "<your GCP private key>",
  GCP_PROJECT_ID: "<your project id>",
  GCP_LOCATION: "<your location>",
  GCP_KEY_RING: "<your key ring>",
  GCP_KEY_NAME: "<your key name>",
  GCP_KEY_VERSION: "<your key version>",
};

/*
check if credentials object contains placeholder values
**/
function checkForPlaceholders() {
  const errorBuffer = Array();
  const placeholderPattern = /^<.*>$/;
  for (const [key, value] of Object.entries(credentials)) {
    // check for placeholder text
    if (`${value}`.match(placeholderPattern)) {
      errorMessage = `You must fill out the ${key} field of your credentials object.`;
      errorBuffer.push(errorMessage);
    }
    // check if value is empty
    else if (value == undefined) {
      error_message = `The value for ${key} is empty. Please enter something for this value.`;
    }
  }
  // raise an error if errors in buffer
  if (errorBuffer.length > 0) {
    message = errorBuffer.join("\n");
    throw message;
  }
}

module.exports = { getCredentials };
